File: python-sdk/src/boxsdk/models/shared/signrequestprefilltag.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from boxsdk import utils


@dataclass_json
@dataclasses.dataclass
class SignRequestPrefillTag:
    r"""SignRequestPrefillTag
    Prefill tags are used to prefill placeholders with signer input data. Only
    one value field can be included.
    """
    
    checkbox_value: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('checkbox_value') }})
    date_value: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('date_value'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    document_tag_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('document_tag_id') }})
    text_value: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('text_value') }})
    

File: sdks/java/lib/src/main/java/models/shared/SignRequestPrefillTag.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * SignRequestPrefillTag
 * Prefill tags are used to prefill placeholders with signer input data. Only
 * one value field can be included.
**/
public class SignRequestPrefillTag {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("checkbox_value")
    public Boolean checkboxValue;
    public SignRequestPrefillTag withCheckboxValue(Boolean checkboxValue) {
        this.checkboxValue = checkboxValue;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("date_value")
    public OffsetDateTime dateValue;
    public SignRequestPrefillTag withDateValue(OffsetDateTime dateValue) {
        this.dateValue = dateValue;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("document_tag_id")
    public String documentTagId;
    public SignRequestPrefillTag withDocumentTagId(String documentTagId) {
        this.documentTagId = documentTagId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("text_value")
    public String textValue;
    public SignRequestPrefillTag withTextValue(String textValue) {
        this.textValue = textValue;
        return this;
    }
}